packages/protocol/contracts/layer1/based/LibProving.sol

```diff
@@ -336,12 +338,12 @@ library LibProving {
                 assert(ts.validityBond == 0 && ts.contester == address(0));
 
                 ts.prover = msg.sender;
-                ts.blockHash = ctx_.tran.blockHash;
-                ts.stateRoot = ctx_.tran.stateRoot;
+                ts.blockHash = pendingTs.blockHash;
+                ts.stateRoot = pendingTs.stateRoot;
 
                 emit TransitionProvedV2({
                     blockId: local.blockId,
-                    tran: ctx_.tran,
+                    tran: pendingTs,
```

Comment:
so @smtmfft the issue is that in the events `stateRoot = 0`?

Reply:
yes, here this ctx_.tran has 0 state because of the non-sync block check before verification.